cylc/flow/network/resolvers.py

```diff
@@ -82,116 +86,184 @@ def collate_workflow_atts(workflow):
     """Collate workflow filter attributes, setting defaults if non-existent."""
     # Append new atts to the end of the list,
     # this will retain order used in index access.
-    return [
-        workflow.owner,
-        workflow.name,
-        workflow.status,
-    ]
+    return {
+        'user': workflow.owner,
+        'workflow': workflow.name,
+        'workflow_sel': workflow.status,
+    }
 
 
-def workflow_ids_filter(w_atts, items):
+def uniq(iterable):
+    """Return a unique collection of the provided items.
+
+    Useful for unhachable things like dicts, relies on __eq__ for teting
+    equality.
+
+    Examples:
+        >>> uniq([1, 1, 2, 3, 5, 8, 1])
+        [1, 2, 3, 5, 8]
+
+    """
+    ret = []
+    for item in iterable:
+        if item not in ret:
+            ret.append(item)
+    return ret
```

Comment:
Although couldn't this be fulfilled using `set(iterable)`?

Reply:
That wouldn't preserve order.